pkgs/build-support/prefer-remote-fetch/default.nix

```
@@ -22,6 +22,8 @@ self: super: {
   fetchrepoproject = args: super.fetchrepoproject ({ preferLocalBuild = false; } // args);
   fetchs3 = args: super.fetchs3 ({ preferLocalBuild = false; } // args);
   fetchsvn = args: super.fetchsvn ({ preferLocalBuild = false; } // args);
-  fetchurl = args: super.fetchurl ({ preferLocalBuild = false; } // args);
+  fetchurl = args: super.fetchurl (super.lib.extends
+    (super.lib.toExtension { preferLocalBuild = args.preferLocalBuild or false; })
+    (super.lib.toFunction args));
```

Sorry, I forgot to consider that `args` might be a function when writing `args.prefenLocalBuild or false`.
It should be

```suggestion
  fetchurl = fpArgs: super.fetchurl (super.lib.extends
    (finalAttrs: args: { preferLocalBuild = args.preferLocalBuild or false; })
    (super.lib.toFunction fpArgs));
```

Or,

```suggestion
  fetchurl = lib.extendMkDerivation {
    constructDrv = previous.fetchurl;
    extendDrvArgs = finalAttrs: args: { prefetchLocalBuild = args.prefetchLocalBuild or false; };
  }
```

The latter handles the `__functionArgs` automatically.

---

The 2nd failed with:

```
error:
       … while calling the 'head' builtin
         at /home/sandro/src/nixpkgs/lib/attrsets.nix:1696:13:
         1695|           if length values == 1 || pred here (elemAt values 1) (head values) then
         1696|             head values
             |             ^
         1697|           else

       … while evaluating the attribute 'value'
         at /home/sandro/src/nixpkgs/lib/modules.nix:1118:7:
         1117|     // {
         1118|       value = addErrorContext "while evaluating the option `${showOption loc}':" value;
             |       ^
         1119|       inherit (res.defsFinal') highestPrio;

       … while evaluating the option `system.build.toplevel':

       … while evaluating definitions from `/nix/store/p5jrkni7q6pa7ckfyqrrjysqh8i5nmdf-source/nixos/modules/system/activation/top-level.nix':

       … while evaluating the option `system.activationScripts':

       … while evaluating definitions from `/nix/store/6mis9m031ia3c7fmxh3gqrq4782dmppn-source/modules/ssh.nix':

       (stack trace truncated; use '--show-trace' to show the full, detailed trace)

       error: function 'extendDrvArgs' called with unexpected argument 'prefetchLocalBuild'
       at /home/sandro/src/nixpkgs/pkgs/build-support/fetchurl/default.nix:72:5:
           71|     finalAttrs:
           72|     {
             |     ^
           73|       # URL to fetch.
```

---

> The 2nd failed with:

That's due to a typo. It should be `preferLocalBuild` instead of `prefetchLocalBuild`.